app/client/views/residencies/forms/edit/edit.js

```diff
@@ -7,39 +7,94 @@ Template.editResidencyModal.onCreated(function () {
 
   // reactive placeholder for home select options with groups
   templateInstance.homeSelectOptionsWithGroups = new ReactiveVar();
+  templateInstance.addNewResidencyClicked = new ReactiveVar(false);
 
-  Meteor.call('getAllResidentSelectOptions', (error, residentIdSelectOptions) => {
-    // update the resident ID select options reactive variable
-    templateInstance.residentIdSelectOptions.set(residentIdSelectOptions);
-  });
+  Meteor.call(
+    'getAllResidentSelectOptions',
+    (error, residentIdSelectOptions) => {
+      // update the resident ID select options reactive variable
+      templateInstance.residentIdSelectOptions.set(
+        residentIdSelectOptions
+      );
+    }
+  );
 
   // get home select options with groups from server
-  Meteor.call('getHomeSelectOptionsWithGroups', (error, homeSelectOptionsWithGroups) => {
-    // update reactive variable with home select options
-    templateInstance.homeSelectOptionsWithGroups.set(homeSelectOptionsWithGroups)
-  });
+  Meteor.call(
+    'getHomeSelectOptionsWithGroups',
+    (error, homeSelectOptionsWithGroups) => {
+      // update reactive variable with home select options
+      templateInstance.homeSelectOptionsWithGroups.set(
+        homeSelectOptionsWithGroups
+      );
+    }
+  );
 });
 
 Template.editResidencyModal.helpers({
-  today () {
+  today() {
     // Default date today, as a string
     return Date();
   },
-  residency () {
+  residency() {
     const templateInstance = Template.instance();
 
     if (templateInstance.data.residency) {
       return templateInstance.data.residency;
     }
   },
-  residentIdOptions () {
+  newResidency() {
+    const templateInstance = Template.instance();
+
+    if (templateInstance.data.residency) {
+      const {
+        residentId,
+        moveIn,
+        _id,
+      } = templateInstance.data.residency;
+      /* If movin is after today move out and hence the movein for next
```

Comment:
What use-case does this cover? Is there a particular situation this is trying to prevent? Pardon my misunderstanding, so lets just clarify this a bit better in the documentation (comment).

Reply:
Normally while ending the current residency the move out date is today's date/ the current date but if, the move in date happens to be after the current date (an unlikely scenario but possible) then it must end on the moveIn date instead of the current date